code/modules/mining/mine_items.dm

```diff
@@ -428,3 +431,20 @@
 	anchored = 1
 	layer = MOB_LAYER - 0.2
 	density = 0
+
+/obj/item/weapon/emptysandbag
+	name = "empty sandbag"
+	desc = "A bag to be filled with sand."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "sandbag"
+	w_class = 1
+
+/obj/item/weapon/emptysandbag/attackby(obj/item/W, mob/user, params)
+	if(istype(W,/obj/item/weapon/ore/glass || /obj/item/weapon/ore/glass/basalt))
+		user << "<span class='notice'>You fill the sandbag.</span>"
+		var/obj/item/stack/sheet/mineral/sandbags/I = new /obj/item/stack/sheet/mineral/sandbags
+		user.unEquip(src)
+		user.put_in_hands(I)
+		qdel(W)
+		qdel(src)
+		return
```

Comment:
please make it end with "else return ..()" so that trashbag can pick it up, among other things.